### PR TITLE
Don't skip doclint

### PR DESCRIPTION
**Pull request by NotMyFault**
The build as-is fails for me locally if I invoke `mvn clean verify -DskipTests`, and should fail on ci.j too, because we're lacking https://github.com/jenkinsci/acceptance-test-harness/pull/890.

### CODE REVIEW

**Review by timja — APPROVED** [review]

File: pom.xml [hunk 1]
```diff
@@ -485,6 +486,8 @@
         <artifactId>maven-source-plugin</artifactId>
         <executions>
           <execution>
+            <phase>compile</phase>
```
Comment (jtnord) [inline pom.xml:489]: Why compile, this will be run on everysplit ATH.

Similar comment to below and the javadoc

File: pom.xml [hunk 2]
```diff
@@ -495,15 +498,13 @@
...
-          <additionalparam>-Xdoclint:none</additionalparam>
-        </configuration>
         <executions>
           <execution>
-            <id>attach-javadocs</id>
-            <configuration>
-              <additionalparam>-Xdoclint:none</additionalparam>
-            </configuration>
+            <phase>compile</phase>
+            <id>generate-javadocs</id>
```
Comment (jtnord) [inline pom.xml:504]: This now IIUC means we generate javadocs twice during the build pipeline,
Once with the parent invocation bound to package and now here?

This also means we create the javadoc on each ATH split.  In the grand scheme of the ATH times this is probably lost but would we just be better off with a `mvn verify -DskipTests` in a branch on the pipeline?

**Review by jtnord — COMMENTED** [review]
Works, but would seem to be inefficient?

File: pom.xml (hunk 2 above)
Comment (timja) [reply pom.xml:504]: see also https://github.com/jenkinsci/acceptance-test-harness/pull/893

Comment (timja) [reply pom.xml:504]: `mvn verify -DskipTests` doesn't run javadoc currently by the looks of it

Comment (timja) [reply pom.xml:504]: https://github.com/jenkinsci/acceptance-test-harness/pull/893 works now

**Comment by timja** [comment]
Is this needed after https://github.com/jenkinsci/acceptance-test-harness/pull/893?

**Comment by NotMyFault** [comment]
> Is this needed after #893?

Given, the new task runs separately to the maven workflow to verify results independantly.

**Pull request closed without merging**